File: offgame-ui/src/app/model/PlanetItem.ts
export class PlanetItem {

    id!:number
    name!:string
    imageName!:string
    usedFields!:number
    totalFields!:number
    leftFields: number
    minTemperature!:number
    maxTemperature!:number
    galaxy!: number
    system!: number
    position!: number

    constructor(data: PlanetItem) {
        Object.assign(this, data);

        this.leftFields = this.totalFields - this.usedFields;
    }

    getFullName() {
        return `${this.name} [${this.galaxy}:${this.system}:${this.position}]`
    }
}

File: offgame-core/src/main/java/org/retal/offgame/dto/PlanetItem.java
package org.retal.offgame.dto;

import lombok.*;

@Getter
@Setter
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class PlanetItem {

    private Long id;
    private String name;
    private String imageName;
    private Long usedFields;
    private Long totalFields;
    private Long minTemperature;
    private Long maxTemperature;
    private Long galaxy;
    private Long system;
    private Long position;
}
